connector-pagerduty/src/main/java/com/redhat/cloud/notifications/connector/pagerduty/PagerDutyCloudEventDataExtractor.java

```diff
@@ -0,0 +1,90 @@
+package com.redhat.cloud.notifications.connector.pagerduty;
+
+import com.redhat.cloud.notifications.connector.CloudEventDataExtractor;
+import com.redhat.cloud.notifications.connector.authentication.AuthenticationDataExtractor;
+import io.vertx.core.json.JsonObject;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.apache.camel.Exchange;
+import org.apache.camel.http.common.HttpMethods;
+import org.apache.commons.validator.routines.UrlValidator;
+import org.apache.http.ProtocolException;
+
+import java.util.MissingResourceException;
+
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.TARGET_URL;
+import static com.redhat.cloud.notifications.connector.pagerduty.ExchangeProperty.ACCOUNT_ID;
+import static com.redhat.cloud.notifications.connector.pagerduty.ExchangeProperty.TARGET_URL_NO_SCHEME;
+import static org.apache.commons.validator.routines.UrlValidator.ALLOW_LOCAL_URLS;
+
+@ApplicationScoped
+public class PagerDutyCloudEventDataExtractor extends CloudEventDataExtractor {
+
+    public static final String NOTIF_METADATA = "notif-metadata";
+    public static final String AUTHENTICATION = "authentication";
+    public static final String METHOD = "method";
+    public static final String TRUST_ALL = "trustAll";
+    public static final String URL = "url";
+
+    public static final String PAYLOAD = "payload";
+    public static final String CUSTOM_DETAILS = "custom_details";
+    public static final String EVENT_ACTION = "event_action";
+
+    private static final UrlValidator HTTPS_URL_VALIDATOR = new UrlValidator(new String[]{"https"}, ALLOW_LOCAL_URLS);
+    private static final UrlValidator HTTP_URL_VALIDATOR = new UrlValidator(new String[]{"http"}, ALLOW_LOCAL_URLS);
+
+    @Inject
+    AuthenticationDataExtractor authenticationDataExtractor;
+
+    @Override
+    public void extract(Exchange exchange, JsonObject cloudEventData) throws IllegalArgumentException, ProtocolException {
+
+        validatePayload(cloudEventData);
+
+        JsonObject customDetails = cloudEventData.getJsonObject(PAYLOAD).getJsonObject(CUSTOM_DETAILS);
+        exchange.setProperty(ACCOUNT_ID, customDetails.getString("account_id"));
+
+        JsonObject metadata = cloudEventData.getJsonObject(NOTIF_METADATA);
+        exchange.setProperty(TARGET_URL, metadata.getString(URL));
+        validateTargetUrl(exchange);
+        exchange.setProperty(TARGET_URL_NO_SCHEME, exchange.getProperty(TARGET_URL, String.class).replace("https://", ""));
+        exchange.setProperty(TRUST_ALL, Boolean.valueOf(metadata.getString("trustAll")));
+
+        JsonObject authentication = metadata.getJsonObject(AUTHENTICATION);
+        authenticationDataExtractor.extract(exchange, authentication);
+
+        cloudEventData.remove(NOTIF_METADATA);
+
+        exchange.getIn().setBody(cloudEventData.encode());
+        exchange.getIn().setHeader(Exchange.HTTP_METHOD, HttpMethods.valueOf(metadata.getString(METHOD)));
```

Comment:
Since there is no other way to send requests to PagerDuty but using the `POST` method, I think we can use the `HttpMethods.POST` constant here.

Reply:
Would it be preferable to set the constant here, or to set it in the Endpoint entry (linked at [the end of my other comment](https://github.com/RedHatInsights/notifications-backend/pull/2869#discussion_r1709508392))

Reply:
If we are not going to use any other HTTP method I think you can definitely leave it as a constant in the connector. We have something similar for Splunk, where we directly specify the HTTP method for the call:

https://github.com/RedHatInsights/notifications-backend/blob/fab15dcff93f61b1fb8f940e2abe1826e073144e/connector-splunk/src/main/java/com/redhat/cloud/notifications/connector/splunk/SplunkRouteBuilder.java#L66-L74